docs/development.md

```diff
@@ -0,0 +1,74 @@
+# Development Environment
+
+This document describes the agreed-upon environment for development. For ease of maintenance, it does not aim to be a
+click by click description, and it assumes familiarity with Windows, Linux, and search engines.
+
+The standard environment is:
+1. Windows
+2. Windows Subsystem for Linux 2 (WSL)
+3. Ubuntu latest in WSL
+4. Visual Studio Code (VS Code)
+
+Other configurations are not prohibited, but keep in mind that when problems arise there will be less support from the
+local community.
+
+## Installation
+
+### 1. Windows
+
+Should be installed for you - easy!
+
+### 2. Windows Subsystem for Linux 2 (WSL)
+
+Follow Microsoft's instructions.
+
+### 3. Ubuntu latest in WSL
+
+#### Ubuntu
+
+- Install Ubuntu latest and open a shell.
+- `sudo apt update`
+- `sudo apt upgrade`
+- `sudo apt install build-essential` to install a compiler needed for Python compilation.
+- `sudo apt install libbz2-dev libffi-dev libreadline-dev libsqlite3-dev libssl-dev zlib1g-dev` to go with the above.
+
+#### Python
+
+- `curl https://pyenv.run | bash`, then follow the instructions to add it to your load path.
+- `pyenv install 3.7.6`
+- `pyenv global 3.7.6` to set it as the default version.
+- `sudo apt install python3-pip`
+- `pip install --upgrade pip`
+- `sudo apt install python3.7-venv`
+
+#### Node.js for npm
+
+- `curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.2/install.sh | bash`
+- `nvm install node`
+
+#### Git
+
+- `git config --global user.name "First Last"`
+- `git config --global user.email "First.Last@gov.bc.ca"`
+- `git config --global credential.helper
+"/mnt/c/Program\ Files/Git/mingw64/libexec/git-core/git-credential-manager.exe"`
```

Comment:
Is there any opposition to using ssh key pairs? Not saying it should be the default, but that's how I am currently accessing github from the command line

Reply:
Interesting, so that's basically just the same as a token, but in the other direction. Plus it doesn't involve installing the Github Credential Manager (and all the other steps I didn't document because I don't remember what they are). I do prefer the pure-linux solution over some Windows bolt-on, and probably translates well to Macs.

If there are no objections, using an SSH seems like a nice way to go.

Reply:
Wait... as long as it works with the VS Code GUI for pushing.

Reply:
It's pretty slick and I believe (?) it took faster/was simpler to setup than the github cred manager.  I can add those steps as well, dependent upon testing that it works with VS Code GUI for pushing which:
- I think it will work? I have i just have been using "git push [commands ]" from the command line within VS code. I will test.